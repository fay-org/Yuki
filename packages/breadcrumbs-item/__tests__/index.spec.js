import { mount } from '@vue/test-utils'
import BreadcrumbsItem from '../index'
import Breadcrumbs from '../../breadcrumbs'

describe('BreadcrumbsItem', () => {
  it('should render as element', () => {
    const breadcrumbsItem = mount(
      <Breadcrumbs>
        <BreadcrumbsItem />
      </Breadcrumbs>,
    )
    expect(() => breadcrumbsItem.unmount()).not.toThrow()
  })
  it('when component have not to or href,it should render as span', () => {
    const wrapper = mount(
      <Breadcrumbs>
        <BreadcrumbsItem />
      </Breadcrumbs>,
    )
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.find('.fect-breadcrumbs__item').html()).toBe(
      '<span class="fect-breadcrumbs__item"><!----><div class="fect-breadcrumbs__separator">/</div></span>',
    )
  })
})
