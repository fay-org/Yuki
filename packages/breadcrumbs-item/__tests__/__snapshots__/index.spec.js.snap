// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbsItem when component have not to or href,it should render as span 1`] = `<nav class="fect-breadcrumbs" style="font-size: 16px;"><span class="fect-breadcrumbs__item"><!----><div class="fect-breadcrumbs__separator">/</div></span></nav>`;
